spring-security-oauth2/src/main/java/org/springframework/security/oauth2/provider/token/DefaultTokenServices.java

```diff
@@ -143,6 +143,11 @@ public OAuth2AccessToken refreshAccessToken(String refreshTokenValue, TokenReque
 		}
 
 		OAuth2Authentication authentication = tokenStore.readAuthenticationForRefreshToken(refreshToken);
+
+		if (authentication == null) {
```

Comment:
This does not seem like a valid scenario. If the `OAuth2RefreshToken` exists so should the `OAuth2Authentication`. How is it possible that `OAuth2Authentication` can be null?